src/ServiceControl.Monitoring/Infrastructure/EndpointInputQueue.cs

```diff
@@ -13,20 +13,7 @@ public EndpointInputQueue(string endpointName, string inputQueue)
         public string EndpointName { get; }
         public string InputQueue { get; }
 
-        public bool Equals(EndpointInputQueue other)
-        {
-            if (other is null)
-            {
-                return false;
-            }
-
-            if (ReferenceEquals(this, other))
-            {
-                return true;
-            }
-
-            return string.Equals(EndpointName, other.EndpointName) && string.Equals(InputQueue, other.InputQueue);
-        }
+        public bool Equals(EndpointInputQueue other) => string.Equals(EndpointName, other.EndpointName) && string.Equals(InputQueue, other.InputQueue);
 
         public override bool Equals(object obj)
```

Comment:
Fancy enough this could be simplified with pattern matching

```
public override bool Equals(object obj) => obj is EndpointInputQueue inputQueue && Equals(inputQueue);
```

similar on the other struct.

But now I wonder if we should just use the default value tuple to store things because that would do all of those things for us. So basically replace `EndpointInputQueue` with (string endpoint, string inputQueue) and simply reference System.ValueTuple

Reply:
If we stick to those custom tuples the GetHashCode could leverage value tuples like

```
public override int GetHashCode() => (EndpointName, InputQueue).GetHashCode()
```

Reply:
I like the simplified versions of `Equal` and `GetHashCode`. 

I would keep the named structs instead of tuple especially for `EndpointMessageType` as it would become `(string, string)` and could lead to ending up in the wrong order of arguments when doing an instantiation.